utils/ips/claimed_ip_port.go

```diff
@@ -32,12 +32,34 @@ type ClaimedIPPort struct {
 	// actually claimed by the peer in question, and not by a malicious peer
 	// trying to get us to dial bogus IPPorts.
 	Signature []byte
-	// The txID that added this peer into the validator set
-	TxID ids.ID
+
+	initOnce sync.Once
+	nodeID   ids.NodeID
+	gossipID ids.ID
+}
+
+func (i *ClaimedIPPort) NodeID() ids.NodeID {
+	i.initOnce.Do(i.init)
+	return i.nodeID
+}
+
+func (i *ClaimedIPPort) GossipID() ids.ID {
+	i.initOnce.Do(i.init)
+	return i.gossipID
 }
 
 // Returns the length of the byte representation of this ClaimedIPPort.
 func (i *ClaimedIPPort) BytesLen() int {
-	// See wrappers.PackPeerTrackInfo.
 	return baseIPCertDescLen + len(i.Cert.Raw) + len(i.Signature)
 }
+
+func (i *ClaimedIPPort) init() {
```

Comment:
Can we make a `New` function and move all of this initialization into there to avoid having to introduce the `sync.Once`?